packages/client/src/components/Login.re

```diff
@@ -0,0 +1,125 @@
+module LoginMutationConfig = [%graphql
+  {|
+
+  mutation LoginMutation($username: RuleWrapper!, $password: String!) {
+    login(username: $username, password: $password) {
+      id
+      token
+      username
+    }
+  }
+
+|}
+];
+
+module LoginMutation = ReasonApolloHooks.Mutation.Make(LoginMutationConfig);
+
+type loginFormState = {
+  hasError: bool,
+  errorMessage: string,
+  isLoading: bool,
+};
+
+type loginFormActions =
+  | SetError(string)
+  | UnsetError
+  | ToggleLoading(bool);
+
+let initialState: loginFormState = {
+  hasError: false,
+  errorMessage: "",
+  isLoading: false,
+};
+
+[@react.component]
+let make = (~onLogin) => {
+  let (state, dispatch) =
+    React.useReducer(
+      (state, action) =>
+        switch (action) {
+        | SetError(errorMessage) => {...state, hasError: true, errorMessage}
+        | UnsetError => {...state, hasError: false, errorMessage: ""}
+        | ToggleLoading(isLoading) => {...state, isLoading}
+        },
+      initialState,
+    );
+
+  let (loginMutation, loginResponse, _f) = LoginMutation.use();
+
+  React.useEffect1(
+    () => {
+      switch (loginResponse) {
+      | Data(payload) =>
+        let authPayload = Auth.Payload.make(payload);
+        dispatch(ToggleLoading(false));
+        onLogin(authPayload);
+      | Error(error) =>
+        dispatch(ToggleLoading(false));
+        dispatch(SetError(error##message));
+      | Loading => dispatch(ToggleLoading(true))
+      | NoData => dispatch(ToggleLoading(false))
+      | Called => ()
+      };
+
+      None;
+    },
+    [|loginResponse|],
+  );
+
+  let login = (username, password) => {
+    loginMutation(
+      ~variables=
+        LoginMutationConfig.make(~username, ~password, ())##variables,
+      (),
+    );
+  };
+
+  let handleSubmit = event => {
+    event->ReactEvent.Synthetic.preventDefault;
+    let formData = ReactEvent.Form.target(event);
+    let username = formData##username##value;
+    let password = formData##password##value;
+
+    ignore(login(username, password));
+  };
+
+  <div
+    className="flex fixed bg-gray-600 w-full min-h-screen z-50 items-center justify-center">
+    <div className="w-full max-w-xs">
+      <form
+        className="bg-white shadow-md rounded px-8 pt-6 pb-8 mb-4"
+        onSubmit=handleSubmit>
+        {state.hasError ? state.errorMessage->React.string : React.null}
+        <div className="mb-4">
+          <label className="block text-gray-700 text-sm font-bold mb-2">
+            "Username"->React.string
+          </label>
+          <input
```

Comment:
These seem quite customised, could be useful to abstract them to separate files later!

Reply:
Leaving this for now until we have some design